nvm.sh

```diff
@@ -12,6 +12,13 @@
 
 NVM_SCRIPT_SOURCE="$_"
 
+NVM_CURRENT_COLOR='0;32m'
+NVM_INSTALLED_COLOR='0;34m'
+NVM_LTS_COLOR='1;33m'
+NVM_NOT_INSTALLED_COLOR='0;31m'
+NVM_DEFAULT_COLOR='0;37m'
+NVM_SYSTEM_COLOR='0;33m'
```

Comment:
do all 6 of these need to be a global env var? it'd be really nice to store this info in a single env var that nvm expanded as needed, like a bitmask or a string or something.

Reply:
These are not exported, are they still considered global env vars? We set these vars to be available to nvm ls, nvm ls-remote and nvm alias if the user doesn't use set-colors. We have a function called nvm_get_colors that breaks up our exported env (NVM_COLORS) and sets these vars based on what is stored in NVM_COLORS. Would it be better instead of having these variables available to the whole file to make them local to each function that needs them are parse out their values in those functions ?

Reply:
if they don't have a `local` declaration, i believe they'll leak into the user's session in some shells (since if they're not function-local, they're global), so if they're only needed locally, adding `local foo` before each foo should address it.

I'm not 100% sure what the best approach is here. It would be unfortunate to have to explode out `NVM_COLORS` manually at each usage site, but it seems worse to me to add 6 env vars instead.

Reply:
We made the variable local to the nvm_print_formated_alias and nvm_print_version functions.